code/_onclick/item_attack.dm

```diff
@@ -25,7 +25,17 @@
 	I.attack(src, user)
 
 /mob/living/proc/attacked_by(obj/item/I, mob/living/user, def_zone)
-	apply_damage(I.force, I.damtype, def_zone)
+	var/dmgcheck = apply_damage(I.force, I.damtype, def_zone)
+	if(!dmgcheck)
+		visible_message("<span class='danger'>[user] has attempted to attack [src] with [I].</span>",
+		"<span class='userdanger'>[user] has attempted to attack [src] with [I]!</span>")
+		return 0
+
+	if (I.hitsound && I.force > 0) //If an item's hitsound is defined and the item's force is greater than zero...
```

Comment:
if(force) will return 1 if force is not 0 if i'm not mistaken , force shouldn't be negative.